### ISSUE TITLE
Optional property documented as required

### ISSUE THREAD

**Issue by ernesto-jimenez** [interagent/prmd#177]
# How to reproduce

Run the following script which sets a resource with an optional (```resource.optional```) and a required (```resource.name```) parameter when creating the resource.

```bash
#!/usr/bin/env bash

echo "\`\`\`"
rm -rf schemata meta.json schema.json README.md
mkdir schemata

echo "---> Prepare"
# Create resource with broken $ref
echo '{
  "description": "Hello world prmd API",
  "id": "hello-prmd",
  "links": [{
  "href": "https://api.hello.com",
  "rel": "self"
  }],
  "title": "Hello Prmd"
}' > meta.json
echo '{
  "$schema": "http://json-schema.org/draft-04/hyper-schema",
  "title": "Resource",
  "description": "description",

  "definitions": {
    "name": {
      "example": "foo",
      "type": "string"
    },
    "optional": {
      "example": "bar",
      "type": "string"
    }
  },

  "links": [
    {
      "description": "Create a new resource.",
      "href": "/resources",
      "method": "POST",
      "rel": "create",
      "schema": {
        "properties": {
          "resource": {
            "properties": {
              "name": {
                "$ref": "/schemata/resource#/definitions/name"
              },
              "optional": {
                "$ref": "/schemata/resource#/definitions/optional"
              }
            },
            "type": "object",
            "required": ["name"]
          }
        },
        "required": ["resource"],
        "type": [
          "object"
        ]
      },
      "title": "Create"
    }
  ],

  "properties": {
    "resource": {
      "properties": {
        "name": {
          "$ref": "/schemata/resource#/definitions/name"
        },
        "optional": {
          "$ref": "/schemata/resource#/definitions/optional"
        }
      },
      "type": "object",
      "required": ["name"]
    }
  },
  "required": ["resource"],
  "type": [
    "object"
  ],
  "id": "schemata/resource"
}' > schemata/resource.json
echo "---> Combine"
prmd combine -m meta.json schemata > schema.json
echo "---> Verify"
prmd verify schema.json && echo "OK"
echo "---> Document"
echo "\`\`\`"
prmd doc schema.json
```
# Expected

```resource.optional``` parameter should be optional.

# Result

```resource.optional``` is required

# Script result
```
---> Prepare
---> Combine
---> Verify
OK
---> Document
```
## Resource
description

### Attributes
<table>
  <tr>
    <th>Name</th>
    <th>Type</th>
    <th>Description</th>
    <th>Example</th>
  </tr>
  <tr>
    <td><strong>resource:name</strong></td>
    <td><em>string</em></td>
    <td></td>
    <td><code>"foo"</code></td>
  </tr>
  <tr>
    <td><strong>resource:optional</strong></td>
    <td><em>string</em></td>
    <td></td>
    <td><code>"bar"</code></td>
  </tr>
</table>

### Resource Create
Create a new resource.

```
POST /resources
```

#### Required Parameters
<table>
  <tr>
    <th>Name</th>
    <th>Type</th>
    <th>Description</th>
    <th>Example</th>
  </tr>
  <tr>
    <td><strong>resource:name</strong></td>
    <td><em>string</em></td>
    <td></td>
    <td><code>"foo"</code></td>
  </tr>
  <tr>
    <td><strong>resource:optional</strong></td>
    <td><em>string</em></td>
    <td></td>
    <td><code>"bar"</code></td>
  </tr>
</table>

#### Curl Example
```bash
$ curl -n -X POST https://api.hello.com/resources \
  -H "Content-Type: application/json" \
  -d '{
  "resource": {
    "name": "foo",
    "optional": "bar"
  }
}'

```

#### Response Example
```
HTTP/1.1 201 Created
```
```json
{
  "resource": {
    "name": "foo",
    "optional": "bar"
  }
}
```

**Comment by geemus** [comment]
Sounds like this should be fixed by #236

**Comment by ernesto-jimenez** [comment]
thanks @ruprict :trophy:
